jdbc/src/main/java/com/interface21/jdbc/core/JdbcTemplate.java

```diff
@@ -14,4 +19,60 @@ public class JdbcTemplate {
     public JdbcTemplate(final DataSource dataSource) {
         this.dataSource = dataSource;
     }
+
+    public void update(final String sql, final PreparedStatementSetter pss) {
+        try (final var conn = dataSource.getConnection(); final var pstmt = conn.prepareStatement(sql)) {
+            log.debug("query : {}", sql);
+            pss.setParameters(pstmt);
+            pstmt.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataAccessException(e);
+        }
+    }
+
+    public void update(final String sql, final Object... parameters) {
+        update(sql, createPreparedStatementSetter(parameters));
+    }
```

Comment:
우와 오버로딩을 이용해서 인자로 들어오는 파라미터들을 pstmt에 값을 주입해주셨네요?! 신기하네요.
굳이 `오버로딩 + 함수 재호출` 을 쓰지 않고도 메서드를 한번 호출해서 파라미터에 값을 주입할 수 있을텐데, 오버로딩을 사용했을 때의 장점이 무엇인지 궁금합니다!
 
`UserDao` 에 `update(final String sql, final PreparedStatementSetter pss)` 를 호출하는 게 없는데 어떻게 작동하는건지 찾느라 시간이 조금 소요된 것 같아요 ㅎㅎ


아래는 그냥 메서드 호출해서 주입한 버전입니다 :)
```suggestion
    public void update(final String sql, final Object... args) {
        try (final var conn = dataSource.getConnection(); final var pstmt = conn.prepareStatement(sql)) {
            setParameters(pstmt, args);
            final var result = pstmt.executeUpdate();
        } catch (SQLException e) { }
    }
```

Reply:
오 꼼꼼히 코드 봐주셨네요 ㅎㅎ
사실 하나의 update 메서드로도 처리 가능하지만 스프링 JdbcTemplate도 오버로딩을 통해 간단한 호출이랑 복잡한 바인딩을 분리했더라구요. 따라서 오버로딩 방식으로 구성해봤습니당